src/main/java/com/jnape/palatable/lambda/monoid/builtin/MapMerge.java

```diff
@@ -0,0 +1,47 @@
+package com.jnape.palatable.lambda.monoid.builtin;
+
+import com.jnape.palatable.lambda.functions.Fn1;
+import com.jnape.palatable.lambda.functions.specialized.BiMonoidFactory;
+import com.jnape.palatable.lambda.functions.specialized.MonoidFactory;
+import com.jnape.palatable.lambda.monoid.Monoid;
+import com.jnape.palatable.lambda.semigroup.Semigroup;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static com.jnape.palatable.lambda.monoid.Monoid.monoid;
+
+public class MapMerge<A, B> implements BiMonoidFactory<Supplier<Map<A, B>>, Semigroup<B>, Map<A, B>> {
```

Comment:
I don't have a dramatically better name here either. I think the challenge is that the ability to merge is a concept, which could be modeled, but isn't. In the meantime, why don't we call this MergeMaps, so that it's clear we're not saying "map then merge".

Reply:
(and of course, update the static factory methods as well)